### PR TITLE
chore(Auth) - Disable social login buttons and show tooltip on preview

### PR DESCRIPTION
**Pull request by SaraVieira**
closes https://github.com/webstudio-is/webstudio-designer/issues/96
closes https://github.com/webstudio-is/webstudio-designer/issues/95
closes https://github.com/webstudio-is/webstudio-designer/issues/39

### CODE REVIEW

File: app/auth/login.tsx [hunk 1]
```diff
@@ -30,6 +32,25 @@ export const links: LinksFunction = () => {
 export const meta: MetaFunction = () => {
   return { title: "Webstudio Login" };
 };
+const isPreviewDeployment = env?.VERCEL_ENV !== "preview";
```
Comment (kof) [inline app/auth/login.tsx:35]: env is not optional, its always an object, right?

Comment (kof) [reply app/auth/login.tsx:35]: also I don't get the logic, it says its not preview but the variable isPreview

Comment (SaraVieira) [reply app/auth/login.tsx:35]: That was for testing, need to revert when it's working

Comment (SaraVieira) [reply app/auth/login.tsx]: Fixed both

File: app/auth/login.tsx [hunk 2]
```diff
@@ -30,6 +32,25 @@ export const links: LinksFunction = () => {
 export const meta: MetaFunction = () => {
   return { title: "Webstudio Login" };
 };
+const isPreviewDeployment = env.VERCEL_ENV === "preview";
```
Comment (kof) [inline app/auth/login.tsx:35]: `isPreviewEnvironment` better?

File: app/auth/login.tsx [hunk 3]
```diff
@@ -52,12 +73,14 @@ export const Login = ({ errorMessage }: { errorMessage: string }) => {
...
           <Flex gap="2" direction="column" align="center">
             <Form action="/auth/github" method="post">
-              <Button type="submit">
-                <Flex gap="1">
-                  <GithubIcon width="16" />
-                  Login with GitHub
-                </Flex>
-              </Button>
+              <WrapperDisabledTooltip isPreview={isPreviewDeployment}>
+                <Button type="submit" disabled={isPreviewDeployment}>
```
Comment (kof) [inline app/auth/login.tsx:77]: can we just make a GithubButton component and then if its preview you would wrap conditionally into tooltip? That tooltip wouldn't need isPreview prop and would be semantically just always render children.

Currently WrapperDisabledTooltip is not always rendering the children and its a bit hard to follow this logic

File: app/auth/login.tsx [hunk 4]
```diff
@@ -30,6 +32,25 @@ export const links: LinksFunction = () => {
 export const meta: MetaFunction = () => {
   return { title: "Webstudio Login" };
 };
+const isPreviewDeployment = env.VERCEL_ENV === "preview";
+
+const WrapperDisabledTooltip = ({
```
Comment (kof) [inline app/auth/login.tsx:37]: `DisabledButtonTooltip`?

File: app/shared/design-system/components/tooltip.tsx [hunk 5]
```diff
@@ -54,7 +59,7 @@ export const Tooltip = ({
           size="1"
           as="p"
           css={{
-            color: "$loContrast",
+            color: "currentColor",
             lineHeight: multiline ? "20px" : undefined,
```
Comment (kof) [inline app/shared/design-system/components/tooltip.tsx:63]: probably my fault, but we shouldn't use arbitrary values, use the theme

**Review by kof — APPROVED** [review]
approving in advance so you can merge once comments addressed

**Comment by kof** [comment]
The login box looks kinda tiny, not sure what to do about it

File: app/shared/design-system/components/tooltip.tsx (hunk 5 above)
Comment (SaraVieira) [reply app/shared/design-system/components/tooltip.tsx:63]: currentColor will never be a an aribiatry value, it will use the value defined in the parent
https://css-tricks.com/cascading-svg-fill-color/

File: app/auth/login.tsx (hunk 4 above)
Comment (SaraVieira) [reply app/auth/login.tsx]: fixed

File: app/auth/login.tsx (hunk 2 above)
Comment (SaraVieira) [reply app/auth/login.tsx]: fixed

File: app/auth/login.tsx (hunk 3 above)
Comment (SaraVieira) [reply app/auth/login.tsx]: Created a login-button component

File: app/auth/login.tsx [hunk 6]
```diff
@@ -52,21 +47,21 @@ export const Login = ({ errorMessage }: { errorMessage: string }) => {
...
                 <Flex gap="1">
                   <GithubIcon width="16" />
                   Login with GitHub
                 </Flex>
-              </Button>
+              </LoginButton>
             </Form>
-            <Form action="/auth/google" method="post">
-              <Button type="submit" disabled>
+            {/* <Form action="/auth/google" method="post">
```
Comment (kof) [inline app/auth/login.tsx:57]: did you forget to remove the comment?

File: app/shared/design-system/components/tooltip.tsx (hunk 5 above)
Comment (kof) [reply app/shared/design-system/components/tooltip.tsx:63]: I ment the lineHeight

Comment (SaraVieira) [reply app/shared/design-system/components/tooltip.tsx]: Done!

File: app/auth/login.tsx (hunk 6 above)
Comment (SaraVieira) [reply app/auth/login.tsx]: Readded

File: app/auth/components/login-button.tsx [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+export const LoginButton = ({
+  children,
+  isDevLogin = false,
+  enabled,
+  ...props
+}: {
+  children: React.ReactChild;
+  isDevLogin?: boolean;
+  onClick?: () => void;
+  enabled: boolean;
```
Comment (kof) [inline app/auth/components/login-button.tsx:16]:
```suggestion
  isEnabled: boolean;
```

**Comment by kof** [comment]
@natlusyht there is, sending you in dm
